From f2e351c6f84de5d536dab2845555dcf0ec6ef06e Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 27 Jun 2024 15:44:43 +0200
Subject: [PATCH 1032/1695] plat-stm32mp2: conf: enable the trusted UI support
 for stm32mp2

This change makes TUI support with the associated drivers and memory
management available on STM32MP2 platform.

CFG_TUI_FRAME_BUFFER_SIZE_MAX defines the TUI framebuffer max size
(default to 16MB) that must be provisioned in OP-TEE virtual memory
for the TUI frame buffer.

The effective framebuffer location (base address and byte size) is read
from the embedded DTB. The address range must be carved out from the
non-secure DDR (DDR non-secure can use as shared memory). The effective
address range is mapped in OP-TEE core at boot time, from display.c.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fc7b68ac1b6ca2a23a9378eec2cdb97bdcd773fc plat-stm32mp2: conf: enable the trusted UI support for stm32mp2]
Change-Id: I4ad2932cf24544c427d894d42c283a4a1c388fdb
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391783
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index cc4ba5b5f..f25b2be04 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -172,6 +172,7 @@ CFG_STM32_TAMP ?= y
 CFG_STM32_UART ?= y
 CFG_STM32_VREFBUF ?= y
 CFG_STPMIC2 ?= y
+CFG_WITH_TUI ?= y
 
 CFG_DRIVERS_I2C ?= $(CFG_STM32_I2C)
 CFG_REGULATOR_FIXED ?= y
@@ -292,3 +293,13 @@ endif
 ifeq ($(CFG_SCMI_CORTEXM_AGENT),y)
 $(call force,CFG_DRIVERS_MAILBOX,y)
 endif
+
+# Trusted User Interface
+ifeq ($(CFG_WITH_TUI),y)
+$(call force,CFG_DISPLAY,y)
+$(call force,CFG_FRAME_BUFFER,y)
+$(call force,CFG_STM32_LTDC,y)
+# Provision virtual space to fit 10MByte plus the TUI frame buffer
+CFG_TUI_FRAME_BUFFER_SIZE_MAX ?= 0x01000000
+CFG_RESERVED_VASPACE_SIZE ?= (10 * 1024 * 1024 + $(CFG_TUI_FRAME_BUFFER_SIZE_MAX))
+endif
-- 
2.47.2

